### PR TITLE
ENH: add offset_curve as GeoSeries method

### PR DESCRIPTION
**Pull request by JamesGardiner**
Implements the shapely offset_curve method as part of #2010

### CODE REVIEW

File: geopandas/tests/test_geom_methods.py
Comment (martinfleis) [inline geopandas/tests/test_geom_methods.py:1592]: You use `l1` above and `g1` here, I believe incorrectly. 

Also, `l1` does not have any CRS attribute so this check boils down to `None == None`, which does not really check if the CRS is properly preserved.

Comment (JamesGardiner) [reply geopandas/tests/test_geom_methods.py:1592]: I was going off what was done in the `minimum_bounding_circle` tests which also does the same crs comparison resulting in a None == None assertion.

I'll remove the crs check from this test as it's not adding value

**Comment by JamesGardiner** [comment]
@martinfleis do you want to explicitly add shapely parameters to the Geopandas methods, or pass them through as kwargs and point users to the shapely docs?

**Comment by martinfleis** [comment]
Better ton be explicit. We risk getting out of sync but it is easier for a user to see the whole signature rather that fishing in shapely docs which kwargs are allowed.

File: geopandas/_vectorized.py
Comment (martinfleis) [inline geopandas/_vectorized.py:643]:
```suggestion
        return shapely.offset_curve(data, distance=distance, quad_segs=quad_segs, join_style=join_style, mitre_limit=mitre_limit)
```

Can we spell it out to be more future-proof in case of some API changes on the shapely side?

**Review by martinfleis — APPROVED** [review]
Thanks!

**Review by m-richards — APPROVED** [review]
@JamesGardiner thanks for the contribution - and thanks for being patient waiting for the second review! I'm hoping I will have a chance to progress these over the next couple of days.

**Comment by JamesGardiner** [comment]
Thanks for the second review @m-richards - no rush on my end to get these merged

**Pull request merged**
